src/main/java/duke/task/ListItem.java

```diff
@@ -0,0 +1,41 @@
+package duke.task;
+
+/**
+ * the abstract task class as a base for the tasks "deadline", "event" annd "todo"
+ * that requires the subclass to define <code>markAsDone()</code> and <code>getDate()</code>
+ */
+public abstract class ListItem {
+    private final boolean isDone;
+    private final String task;
+
+    public ListItem(String task) {
+        this.task = task;
+        this.isDone = false;
+    }
+
+    public ListItem(String task, boolean isDone) {
+        this.task = task;
+        this.isDone = isDone;
+    }
+
+    /**
+     * @return a boolean of the task's status,
+     * useful for subclasses that passed the status to parent (this class)
+     */
+    public boolean getDone() {
```

Comment:
I recommend to rename this with isDone to be consistent with the coding standards.